### ISSUE TITLE
A type like {(el: HTMLElement) => Promise<void>|void} no longer allowed in 28.0.0

### ISSUE THREAD

**Issue by doberkofler** [gajus/eslint-plugin-jsdoc#592]
## Expected behavior
Recognize the valid type as before in 27.1.2

## Actual behavior
```
warning: Syntax error in type: (el: HTMLElement) => Promise<void>|void (jsdoc/valid-types) at src/framework/module/index.ts:
  76 |  * Specify the function to execute when loading a react page.
  77 |  *
> 78 |  * @param {(el: HTMLElement) => Promise<void>|void} render - A function that will render the page when the DOM is ready.
     | ^
  79 |  * @example newReactPage({console.log('DOM is ready');});
  80 |  */
  81 | export function newReactPage(render: (el: HTMLElement) => Promise<void>|void): void {
````
## ESLint Config

```
'jsdoc/valid-types': 'warn'
```
## ESLint sample

```ts
/**
 * Specify the function to execute when loading a react page.
 *
 * @param {(el: HTMLElement) => Promise<void>|void} render - A function that will render the page when the DOM is ready.
 * @example newReactPage({console.log('DOM is ready');});
 */
export function newReactPage(render: (el: HTMLElement) => Promise<void>|void): void {
}
```

## Environment

* Node version: 14.4.0
* ESLint version 7.3.0
* `eslint-plugin-jsdoc` version: 28.0.0

**Comment by brettz9** [comment]
You now need the mode set in your config if you are going to use such types:
 
```js
settings: {jsdoc: {mode: "typescript"}}
```

**Comment by doberkofler** [comment]
Thank you!

**Comment by PatrickHollweck** [comment]
Sorry for reviving this old issue.
I have the same problem even though I did set the mode to "typescript"

Here is my minimal config:

```javascript
module.exports = {
  root: true,
  extends: [
    "eslint:recommended",
    "plugin:jsdoc/recommended",
  ],
  parser: "babel-eslint",
  parserOptions: {
    ecmaVersion: 9,
    sourceType: "module",
  },
  env: {
    es6: true,
    node: true,
    jquery: true,
    browser: true,
  },
  plugins: ["jsdoc"],
  rules: {
    "jsdoc/require-jsdoc": [0],
    "jsdoc/require-param-description": [0],
    "jsdoc/require-returns-description": [0],
  },
  settings: {
      jsdoc: {
          mode: "typescript"
      }
  }
};
```

**Comment by brettz9** [comment]
Please file a new issue with your details. Also be sure to test against the latest version. Thanks!
